spiketoolkit/validation/quality_metric_classes/noise_overlap.py

```diff
@@ -23,47 +26,75 @@ def __init__(self, metric_data):
         if not metric_data.has_recording():
             raise ValueError("MetricData object must have a recording")
 
-    def compute_metric(self, max_spikes_per_unit_for_noise_overlap, num_features, num_knn, **kwargs):
+    def compute_metric(self, num_channels_to_compare, max_spikes_per_unit_for_noise_overlap,
+                        num_features, num_knn, **kwargs):
+
+        # update keyword arg in case it's already specified to something
+        kwargs['max_spikes_per_unit'] = max_spikes_per_unit_for_noise_overlap
+        kwargs['recompute_info'] = True
         params_dict = update_all_param_dicts_with_kwargs(kwargs)
         save_property_or_features = params_dict['save_property_or_features']
         seed = params_dict['seed']
 
+        # first, get waveform snippets of every unit (at most n spikes)
+        # waveforms = List (units,) of np.array (n_spikes, n_channels, n_timepoints)
         waveforms = st.postprocessing.get_unit_waveforms(
             self._metric_data._recording,
             self._metric_data._sorting,
             unit_ids=self._metric_data._unit_ids,
-            max_spikes_per_unit=max_spikes_per_unit_for_noise_overlap,
             **kwargs
         )
 
+        # set random seed
         if seed is not None:
             np.random.seed(seed)
 
+        # get channel idx and locations
+        channel_idx = np.arange(self._metric_data._recording.get_num_channels())
+        channel_locations = self._metric_data._channel_locations
+
+        if num_channels_to_compare > len(channel_idx):
+            num_channels_to_compare = len(channel_idx)
+
+        # get noise snippets
+        min_time = min([self._metric_data._sorting.get_unit_spike_train(unit_id=unit)[0]
+                    for unit in self._metric_data._sorting.get_unit_ids()])
+        max_time = max([self._metric_data._sorting.get_unit_spike_train(unit_id=unit)[-1]
+                    for unit in self._metric_data._sorting.get_unit_ids()])
+        times_control = np.random.choice(np.arange(min_time, max_time),
```

Comment:
What if every unit has less spikes than `max_spikes_per_unit_for_noise_overlap`? Safe to add:

```
max_spikes = np.max([len(self._metric_data._sorting.get_unit_spike_train(u)) for u in self._metric_data._unit_ids])

if max_spikes < max_spikes_per_unit_for_noise_overlap:
    max_spikes_per_unit_for_noise_overlap = max_spikes
```

Reply:
Added this